libmamba/include/mamba/core/graph_util.hpp

```diff
@@ -547,7 +554,8 @@ namespace mamba
     inline void DiGraph<N, E>::add_edge_impl(node_id from, node_id to, T&& data)
     {
         Base::add_edge(from, to);
-        m_edges[{ from, to }] = std::forward<T>(data);
```

Comment:
this requires the default constructor. In case of a MNode/MEdge with variants as constructor args, the default constructor is deleted.